resources/js/challenge.js

```diff
@@ -0,0 +1,66 @@
+window.addEventListener('load', init);
+
+let container;
+let challengeInfo;
+
+function init() {
+    let list = document.getElementById('challengeList');
+    container = document.getElementById('challengeContainer');
+    let details = document.getElementById('info');
+    list.addEventListener('click', challengeClickHandler);
+    createInfoContainer();
+}
+
+function ajax(url, succesHandler) {
+    fetch(url)
+        // .then((res) => res.json())
+        .then((res) => {
+            return res.json()
+        })
+        .then(succesHandler)
+        .catch((res) => {
+
+        })
+}
+
+function challengeClickHandler(e) {
+    let clickedItem = e.target;
+    if (clickedItem.nodeName !== 'SPAN' && clickedItem.nodeName !== 'BUTTON') {
+        return;
+    }
+    if (challengeInfo){
+        while (challengeInfo.firstChild){
+            challengeInfo.removeChild(challengeInfo.firstChild)
+        }
+    }
+    ajax((apiurl + e.target.dataset.id), showInfo);
+}
+
+function createInfoContainer(data) {
+    console.log(data);
+    let challenge = document.createElement('div');
+    challenge.id = 'infoContainer'
+    challenge.classList.add('col-md-7', 'p-1');
+    container.appendChild(challenge);
+}
+
+function showInfo(data) {
+    challengeInfo = document.getElementById('infoContainer')
```

Comment:
Doe dit misschien in de init functie

Reply:
infoContainer word pas aangemaakt in de createInfoContainer functie dus dat werkt niet.